yt_dlp/extractor/common.py

```diff
@@ -2535,18 +2535,21 @@ def extract_common(source):
                 if segment_duration:
                     ms_info['segment_duration'] = float(segment_duration)
 
+            base_url_elem = element.find(_add_ns('BaseURL'))
+            base_url = None if base_url_elem is None else base_url_elem.text
+
             def extract_Initialization(source):
                 initialization = source.find(_add_ns('Initialization'))
                 if initialization is not None:
-                    ms_info['initialization_url'] = initialization.attrib['sourceURL']
+                    ms_info['initialization_url'] = initialization.attrib.get('sourceURL', base_url)
 
             segment_list = element.find(_add_ns('SegmentList'))
             if segment_list is not None:
                 extract_common(segment_list)
                 extract_Initialization(segment_list)
                 segment_urls_e = segment_list.findall(_add_ns('SegmentURL'))
                 if segment_urls_e:
-                    ms_info['segment_urls'] = [segment.attrib['media'] for segment in segment_urls_e]
+                    ms_info['segment_urls'] = [(segment.attrib.get('media', base_url), segment.attrib.get('mediaRange', None)) for segment in segment_urls_e]
```

Comment:
Same

Reply:
It's recommended (I read) to use `element.get()` since that exists without any possible `dict` construction implied by `element.attrib`. `element.attrib[key]` can be a way of forcing an exception if `key` is missing. The yt-dl code is prone to use `.attrib` unnecessarily, perhaps for prehistorical reasons.